consumer/sm_context.go

```diff
@@ -24,16 +25,67 @@ import (
 	"github.com/omec-project/openapi/models"
 )
 
+func getServingSmfIndex(smfNum int) (servingSmfIndex int) {
+	servingSmfIndexStr := os.Getenv("SERVING_SMF_INDEX")
```

Comment:
Why we are using ENV variable for this ? Each AMF instance can have its own global variable which can do round robin on the list of SMFs.

Reply:
This is temporary solution, with global SMF list via NRF cache changes, this code shall be removed.